pkg/webhook/controlplane/ensurer.go

```diff
@@ -392,18 +397,34 @@ func (e *ensurer) ShouldProvisionKubeletCloudProviderConfig(ctx context.Context,
 
 // EnsureKubeletCloudProviderConfig ensures that the cloud provider config file conforms to the provider requirements.
 func (e *ensurer) EnsureKubeletCloudProviderConfig(ctx context.Context, _ genericmutator.EnsurerContext, data *string, namespace string) error {
-	// Get `cloud-provider-config` ConfigMap
+	secret := &corev1.Secret{}
+	secretErr := e.client.Get(ctx, kutil.Key(namespace, azure.CloudProviderDiskConfigName), secret)
+	if client.IgnoreNotFound(secretErr) != nil {
+		return errors.Wrapf(secretErr, "could not get secret '%s/%s'", namespace, azure.CloudProviderDiskConfigName)
+	}
+
+	if len(secret.Data[azure.CloudProviderConfigMapKey]) > 0 {
+		*data = string(secret.Data[azure.CloudProviderConfigMapKey])
+		return nil
+	}
+
+	if !apierrors.IsNotFound(secretErr) && len(secret.Data[azure.CloudProviderConfigMapKey]) == 0 {
+		return nil
+	}
```

Comment:
Hm, what is this condition for?

Reply:
It's this [case](https://github.com/gardener/gardener-extension-provider-azure/blob/3b7440e45f4905430348e6798d17f487f2a01a66/pkg/webhook/controlplane/ensurer.go#L405-L408) but for the secret instead of the config map.